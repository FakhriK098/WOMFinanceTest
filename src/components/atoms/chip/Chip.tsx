import React from 'react';
import { StyleSheet, View } from 'react-native';
import Text from '@components/atoms/text/Text';
import { spacing } from '@theme/spacing';
import { radius } from '@theme/radius';

type Props = { label: string; color: string; textColor: string };

const Chip = ({ label, color, textColor }: Props) => {
  return (
    <View style={[styles.base, { backgroundColor: color }]}>
      <Text size={12} color={textColor}>
        {label}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  base: {
    paddingHorizontal: spacing.sm,
    paddingVertical: 4,
    borderRadius: radius.md,
    alignSelf: 'flex-start',
  },
});

export default React.memo(Chip);
