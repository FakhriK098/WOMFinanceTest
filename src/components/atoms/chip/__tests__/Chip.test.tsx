import React from 'react';
import renderer, { act } from 'react-test-renderer';

import Chip from '../Chip';

describe('Chip atom', () => {
  it('matches snapshot', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <Chip label="Fire" color="#f00" textColor="#fff" />,
      );
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    act(() => tr!.unmount());
  });

  it('renders label text', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <Chip label="Fire" color="#f00" textColor="#fff" />,
      );
    });
    const json = tr!.toJSON();
    const hasText = (node: any): boolean => {
      if (!node) return false;
      if (typeof node === 'string') return node === 'Fire';
      const children = node.children || [];
      return Array.isArray(children) && children.some(hasText);
    };
    expect(hasText(json)).toBe(true);
    act(() => tr!.unmount());
  });

  it('applies background color style', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <Chip label="X" color="#123456" textColor="#fff" />,
      );
    });
    const wrapper = tr!.root.find((inst) => inst.props && inst.props.style);
    const style = wrapper.props.style as any[];
    const flattened = Array.isArray(style) ? style : [style];
    const hasBg = flattened.some((s) => s && s.backgroundColor === '#123456');
    expect(hasBg).toBe(true);
    act(() => tr!.unmount());
  });
});
