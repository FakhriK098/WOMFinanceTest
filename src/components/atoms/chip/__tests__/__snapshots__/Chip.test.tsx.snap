// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip atom matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 12,
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "#f00",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 12,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    Fire
  </Text>
</View>
`;
